.github/workflows/build.yml

```diff
@@ -137,19 +140,38 @@ jobs:
       - name: Set env
         run: |
           cat tag.txt >> $GITHUB_ENV
+      - name: Create cosign public key
+        run: |
+          echo -n "${{ secrets.COSIGN_PUBLIC_KEY }}" | base64 --decode > cosign.pub
+      - uses: actions/upload-artifact@v2
+        with:
+          name: cosign-public-key
+          path: cosign.pub
       - name: Publish images
         if: github.event_name == 'push'
         run: |
-          LIFECYCLE_IMAGE_TAG=$(git describe --always)
           DOCKER_CLI_EXPERIMENTAL=enabled
-          go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+linux.x86-64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux
-          go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+linux.arm64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux-arm64 -arch arm64
-          go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+windows.x86-64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-windows -os windows
+          LIFECYCLE_IMAGE_TAG=$(git describe --always)
+
+          LINUX_AMD64_SHA=$(go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+linux.x86-64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux | awk '{print $NF}')
+          echo "LINUX_AMD64_SHA: $LINUX_AMD64_SHA"
+
+          LINUX_ARM64_SHA=$(go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+linux.arm64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux-arm64 -arch arm64 | awk '{print $NF}')
+          echo "LINUX_ARM64_SHA: $LINUX_ARM64_SHA"
+
+          WINDOWS_AMD64_SHA=$(go run ./tools/image/main.go -lifecyclePath ./out/lifecycle-v*+windows.x86-64.tgz -tag buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-windows -os windows | awk '{print $NF}')
+          echo "WINDOWS_AMD64_SHA: $WINDOWS_AMD64_SHA"
+
           docker manifest create buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG} \
-              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux \
-              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux-arm64 \
-              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-windows
-          docker manifest push buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}
+              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux@${LINUX_AMD64_SHA} \
+              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-linux-arm64@${LINUX_ARM64_SHA} \
+              buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}-windows@${WINDOWS_AMD64_SHA}
+
+          MANIFEST_SHA=$(docker manifest push buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG})
+          echo "MANIFEST_SHA: $MANIFEST_SHA"
+
+          echo -n "${{ secrets.COSIGN_PASSWORD }}" | cosign sign -key <(echo -n "${{ secrets.COSIGN_PRIVATE_KEY }}" | base64 --decode) -a tag=${LIFECYCLE_IMAGE_TAG} -r buildpacksio/lifecycle:${LIFECYCLE_IMAGE_TAG}@${MANIFEST_SHA}
```

Comment:
`cosign sign` supports reading the password from the env var `COSIGN_PASSWORD`, so this could be `COSIGN_PASSWORD=${{ secrets.COSIGN_PASSWORD }}`, or through `with:` for this step.

Reply:
Updated in bbcfa1c